### PR TITLE
Remove base_image functionality and configgin from fissile

### PR DESCRIPTION
**Pull request by dragonchaser**
This PR removes the base_image functionality from fissile which is no longer needed since the base image is being pulled from dockerhub.
It also removes configgin from fissile because configgin has been gemified by @aduffeck and is provided through the base_image.

**Note:** this is a WIP, do not merge yet

### CODE REVIEW

File: builder/role_image_test.go [hunk 1]
```diff
@@ -441,11 +441,11 @@ func TestBuildRoleImages(t *testing.T) {
 	// Check that making the first wait for the second job works
 	secondJobReady := make(chan struct{})
 	mockBuilder.callback = func(name string) error {
-		if strings.Contains(name, "-myrole:") {
+		if strings.Contains(name, "-myrole-") {
 			<-secondJobReady
 			return nil
 		}
-		if strings.Contains(name, "-foorole:") {
+		if strings.Contains(name, "-foorole-") {
```
Comment (viovanov) [inline builder/role_image_test.go]: what's the name supposed to be? `-` suggests we added some sort of suffix?

File: cmd/build-images.go [hunk 2]
```diff
@@ -67,9 +71,13 @@ from other specs.  At most one is allowed.  Its syntax is --patch-properties-rel
 			flagBuildImagesForce = true
 		}
 
+		if flagBuildImagesStemcell == "" {
```
Comment (viovanov) [inline cmd/build-images.go]: iirc, validation shouldn't happen here, but in `GenerateRoleImages`

File: cmd/build-images.go [hunk 3]
```diff
@@ -126,5 +134,12 @@ func init() {
 		"Output the result as tar files in the given directory rather than building with docker",
 	)
 
+	buildImagesCmd.PersistentFlags().StringP(
+		"stemcell",
+		"",
```
Comment (viovanov) [inline cmd/build-images.go]: `-s` could work

File: cmd/build-images.go [hunk 4]
```diff
@@ -126,5 +134,12 @@ func init() {
...
+		"",
```
Comment (viovanov) [inline cmd/build-images.go]: I think we can have a default - `suse/fissile-opensuse-stemcell` (i think that's what we settled on, right?)

File: cmd/build-packages.go [hunk 5]
```diff
@@ -41,8 +43,12 @@ compiled once.
 			return err
 		}
 
+		if flagBuildPackagesStemcell == "" {
```
Comment (viovanov) [inline cmd/build-packages.go]: same as before about where validation needs to happen - see other commands

File: compilator/compilator.go [hunk 6]
```diff
@@ -797,19 +691,9 @@ func (c *Compilator) getPackageContainerName(pkg *model.Package) string {
...
-
-// baseCompilationImageRepository will return the compilation image repository
-func (c *Compilator) baseCompilationImageRepository() string {
-	return fmt.Sprintf("%s-cbase", c.repositoryPrefix)
-}
-
 // BaseImageName returns the name of the compilation base image
 func (c *Compilator) BaseImageName() string {
-	return util.SanitizeDockerName(fmt.Sprintf("%s:%s", c.baseCompilationImageRepository(), c.baseCompilationImageTag()))
+	return c.stemcellImageName
```
Comment (viovanov) [inline compilator/compilator.go]: we don't need this function anymore, right?

File: compilator/compilator_test.go [hunk 7]
```diff
@@ -245,95 +244,6 @@ func getContainerIDs(imageName string) ([]string, error) {
 	return results, nil
 }
 
-func TestContainerKeptAfterCompilationWithErrors(t *testing.T) {
```
Comment (viovanov) [inline compilator/compilator_test.go]: why have all these tests been removed?

File: docs/fissile_build_images.md [hunk 8]
```diff
@@ -39,6 +39,7 @@ fissile build images
   -O, --output-directory string           Output the result as tar files in the given directory rather than building with docker
   -P, --patch-properties-release string   Used to designate a "patch-properties" psuedo-job in a particular release.  Format: RELEASE/JOB.
       --roles string                      Build only images with the given role name; comma separated.
+      --stemell                           Docker base stemcell (e.g. fissile-stemcell-opensuse:42.2
```
Comment (viovanov) [inline docs/fissile_build_images.md]: typo these can be regenerated
see `fissile docs markdown`

File: builder/packages_image_test.go [hunk 9]
```diff
@@ -18,6 +18,11 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+const (
+	dockerImageEnvVar      = "FISSILE_TEST_DOCKER_IMAGE"
```
Comment (dragonchaser) [inline builder/packages_image_test.go]: Done

File: builder/role_image_test.go (hunk 1 above)
Comment (dragonchaser) [inline builder/role_image_test.go]: Using `:` will potentially result in Docker not being able to create the underlying volumes. This is the reason why we disallowed the usage of `:`.

File: docs/fissile_build_images.md (hunk 8 above)
Comment (dragonchaser) [inline docs/fissile_build_images.md]: Done

File: compilator/compilator.go (hunk 6 above)
Comment (dragonchaser) [inline compilator/compilator.go]: Since fissile version is not part of the ImageName anymore, I do not think so.

File: Makefile [hunk 10]
```diff
@@ -18,7 +18,7 @@ lint:
 vet:
 	${GIT_ROOT}/make/vet
 
-bindata: scripts/configgin/output/configgin.tgz
+bindata:
```
Comment (mook-as) [inline Makefile]: Don't we still need it for the various bash scripts?

File: cmd/build-images.go (hunk 2 above)
Comment (dragonchaser) [inline cmd/build-images.go]: We removed this check since it is obsolete

File: cmd/build-packages.go (hunk 5 above)
Comment (dragonchaser) [inline cmd/build-packages.go]: We removed this check since it is obsolete

File: Makefile (hunk 10 above)
Comment (dragonchaser) [inline Makefile]: removed bindata

File: cmd/build-images.go (hunk 3 above)
Comment (dragonchaser) [inline cmd/build-images.go]: Done

File: cmd/build-images.go (hunk 4 above)
Comment (dragonchaser) [inline cmd/build-images.go]: Done

File: Makefile (hunk 10 above)
Comment (viovanov) [inline Makefile]: @mook-as we need it for `Dockerfile-role`

File: compilator/compilator.go (hunk 6 above)
Comment (viovanov) [inline compilator/compilator.go]: Can we remove it then?

File: make/test [hunk 11]
```diff
@@ -3,4 +3,4 @@
 set -o errexit
 
 go generate ./docker/ # for mocks
-go test -race -cover $(go list -f '{{ .ImportPath }}' ./... | sed '/fissile[/]scripts/d ; /\/vendor\//d')
+go test -cover $(go list -f '{{ .ImportPath }}' ./... | sed '/fissile[/]scripts/d ; /\/vendor\//d')
```
Comment (viovanov) [inline make/test]: we shouldn't check this in

File: scripts/compilation/scripts.go [hunk 12]
```diff
@@ -6,6 +6,8 @@ import (
 )
 
 const (
+	// OpenSUSE ist the name of the openSUSE base image
+	OpenSUSE = "opensuse"
```
Comment (viovanov) [inline scripts/compilation/scripts.go]: are we using these?
i don't think these should be needed anymore

File: util/name_sanitization_test.go [hunk 13]
```diff
@@ -13,7 +13,7 @@ func TestSanitizeDockerName(t *testing.T) {
 		"semver+build":  "semver-build",
 		"branch-name":   "branch-name",
 		"remote/branch": "remote-branch",
-		"image:tag":     "image:tag",
+		"image:tag":     "image-tag",
```
Comment (viovanov) [inline util/name_sanitization_test.go]: this doesn't sound right
we can't not have image tags :/

**Comment by aduffeck** [comment]
Superseded by #233

**Pull request closed without merging**